bc_obps/reporting/service/report_emission_allocation_service.py

```diff
@@ -0,0 +1,214 @@
+    ReportProductEmissionAllocationsSchemaIn,
+    ReportProductEmissionAllocationsSchemaOut,
+)
+from reporting.models.report_product_emission_allocation import ReportProductEmissionAllocation
+from reporting.models.report_operation import ReportOperation
+from reporting.models.report_product import ReportProduct
+from reporting.models import FacilityReport
+from reporting.service.emission_category_service import EmissionCategoryService
+
+
+class ReportEmissionAllocationService:
+    @staticmethod
+    @transaction.atomic()
+    def get_emission_allocation_data(
+        report_version_id: int, facility_id: UUID
+    ) -> ReportProductEmissionAllocationsSchemaOut:
+        # Step 1: Get the facility report ID
+        facility_report_id = FacilityReport.objects.get(report_version_id=report_version_id, facility_id=facility_id).pk
+
+        # Step 2: Fetch all emission category totals
+        all_emission_categories_totals = ReportEmissionAllocationService.get_emission_category_totals_with_ids(
+            facility_report_id
+        )
+
+        # Step 3: Fetch report products for the given report version and facility
+        report_products = ReportProduct.objects.filter(
+            report_version_id=report_version_id, facility_report__facility_id=facility_id
+        )
+
+        # Step 4: Fetch existing emission allocations for the given report version and facility
+        report_product_emission_allocations = ReportProductEmissionAllocation.objects.filter(
+            report_version_id=report_version_id, report_product__facility_report__facility_id=facility_id
+        )
+
+        # Step 5: Construct the response data
+        existing_allocation = (
+            report_product_emission_allocations.first()
+        )  # get the first record (if any) to get the allocation_methodology and allocation_other_methodology_description
+        allocation_methodology = ""  # because these fields are the same for all records of a report
+        allocation_other_methodology_description = ""
+        if existing_allocation is not None:
+            allocation_methodology = existing_allocation.allocation_methodology
+            if existing_allocation.allocation_other_methodology_description is not None:
+                allocation_other_methodology_description = existing_allocation.allocation_other_methodology_description
+
+        report_product_emission_allocations_data = []
+        total_reportable_emissions = 0
+        for category, data in all_emission_categories_totals.items():
+            if (
+                data["type"] == "other_excluded"
+            ):  # these categories cannot currently be allocated to products, so we skip them
+                continue
+            if (
+                data["type"] == "basic"
+            ):  # only these categories are reportable, so we sum them up to get the total reportable emissions
+                total_reportable_emissions += data["total"]
```

Comment:
Is this block necessary? Isn't the total for emissions attributable for reporting already calculated in the EmissionCategoryService?